pkg/apis/intevents/v1alpha1/brokercell_defaults.go

```diff
@@ -19,12 +19,30 @@ package v1alpha1
 import (
 	"context"
 
+	quantityutil "github.com/google/knative-gcp/pkg/utils/resource"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"knative.dev/pkg/ptr"
 )
 
 const (
-	minReplicas int32 = 1
-	maxReplicas int32 = 10
+	avgCPUUtilization int32 = 95
+	// The limit we set (for Fanout and Retry) is 3000Mi which is mostly used
+	// to prevent surging memory usage causing OOM.
+	// Here we only set half of the limit so that in case of surging memory
+	// usage, HPA could have enough time to kick in.
+	// See: https://github.com/google/knative-gcp/issues/1265
+	avgMemoryUsage                         string  = "1500Mi"
+	avgMemoryUsageIngress                  string  = "700Mi"
+	cpuRequest                             string  = "1000m"
+	cpuRequestFanout                       string  = "1500m"
+	cpuLimit                               string  = ""
+	memoryRequest                          string  = "500Mi"
+	memoryLimitToRequestCoefficient        float64 = 6.0
```

Comment:
What's the motivation to use a coefficient rather than an absolute number?

Reply:
Part of this work was accounting for the HPA thresholds and handle the cases when they are not specified explicitly. For instance, if one sets "50Mi" and "300Mi" for requests and limits, respectively, then the original threshold of "1500Mi" may not be as suitable anymore -- instead, we will proportionally adjust it to a more or less suitable value of "150Mi".